packages/flutter/lib/src/material/pickers/input_date_picker.dart

```diff
@@ -131,12 +132,18 @@ class InputDatePickerFormField extends StatefulWidget {
   /// {@macro flutter.widgets.editableText.autofocus}
   final bool autofocus;
 
+  /// Controls the text being edited.
+  ///
+  /// If null, this widget will create its own [TextEditingController] and
+  /// initialize its [TextEditingController.text] with [initialValue].
+  final TextEditingController? controller;
+
   @override
   _InputDatePickerFormFieldState createState() => _InputDatePickerFormFieldState();
 }
 
 class _InputDatePickerFormFieldState extends State<InputDatePickerFormField> {
-  final TextEditingController _controller = TextEditingController();
+  TextEditingController _controller = TextEditingController();
```

Comment:
Why are you creating a `TextEditingController` here? You will create another on `initState` or get the one from the `widget`. This will leak memory as you won't `dispose` it.